### DISCUSSION TITLE
Prevent empty traits from being included in the Code view and/or toHTML result

### DISCUSSION THREAD

**Discussion by protozoo** [GrapesJS/grapesjs#4107] [Q&A]
Hi there,

I have a few components with a text-type trait set to null by default. 

If the user doesn't change it, this trait is not included in the HTML output as an attribute (that's fine!). 
**HTML Output**: `<mycomp />`

If the user edits the trait, setting its value to X, this value is included in the output HTML too (fine again!). 
**HTML Output**: `<mycomp mytrait="X" />`

However, if the user deletes the content of the trait field leaving it empty, the HTML output still includes the attribute with an empty string:
**HTML Output**: `<mycomp mytrait="" />`

Is there any way from preventing the attribute from being. included at all?
**Expected HTML Output**: `<mycomp />`

Many thanks!

**Comment by artf** [comment]
Attributes with `null`, `undefined`, `false` values are not returned in HTML output, but the empty string is valid, so you should probably set a listener and reset those values in the case are empty.
